src/extract.rs

```diff
@@ -351,3 +354,13 @@ fn write_toml<W: std::io::Write>(
 
     Ok(())
 }
+
+fn anchor_prefix(id: &str) -> &str {
+    if let Some(_) = SECTION_ID.captures(id) {
+        &"section-"
+    } else if let Some(_) = APPENDIX_ID.captures(id) {
+        &"appendix-"
+    } else {
+        &""
+    }
```

Comment:
nit: regex seems like overkill here. could we just do:

```suggestion
    match id.chars().next().unwrap_or('_') {
        '0'..='9' => "section-",
        'a'..='z' | 'A'..='Z' => "appendix-",
        _ => ""
    }
```